src/gg.regression.unity.experimental.runespawn/Editor/Scripts/CompilationController.cs

```diff
@@ -0,0 +1,187 @@
+            return;
+        }
+
+        try
+        {
+            List<CompilationMessage> messageCopy;
+
+            // Lock the error list to prevent race conditions
+            lock (errorLock)
+            {
+                messageCopy = new List<CompilationMessage>(compilationErrors);
+            }
+
+            if (messageCopy.Count == 0)
+            {
+                // If no errors, send a success message
+                messageCopy.Add(new CompilationMessage
+                {
+                    File = "",
+                    Line = 0,
+                    Column = 0,
+                    Message = "Compilation finished successfully.",
+                    Type = "Success"
+                });
+            }
+
+            // Send the JSON data to the client
+            Utilities.SendJsonResponse(client.GetStream(), new { status = "Success", messages = messageCopy });
+
+            client.Close();
+        }
+        catch (Exception ex)
+        {
+            Debug.LogError($"CompilationController: Exception while sending compilation errors - {ex.Message}");
+            Utilities.SendJsonResponse(client.GetStream(), new { status = "Error", message = ex.Message });
+            client.Close();
+        }
+    }
+
+    /// <summary>
+    /// Event handler for when an assembly compilation is finished.
+    /// Captures compilation errors.
+    /// </summary>
+    /// <param name="assemblyPath">Path to the assembly.</param>
+    /// <param name="messages">Compilation messages.</param>
+    private static void OnAssemblyCompilationFinished(string assemblyPath, CompilerMessage[] messages)
+    {
+        lock (errorLock)
+        {
+            compilationErrors.Clear();
+
+            foreach (var message in messages)
+            {
+                CompilationMessage compMessage = new CompilationMessage
+                {
+                    File = message.file,
+                    Line = message.line,
+                    Column = message.column,
+                    Message = message.message,
+                    Type = message.type.ToString()
+                };
+                compilationErrors.Add(compMessage);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Event handler for when compilation is finished.
+    /// Sends a response to the compile client.
+    /// </summary>
+    /// <param name="context">Compilation context.</param>
+    private static void OnCompilationFinished(object context)
+    {
+        Debug.Log("CompilationController: Compilation finished.");
+        if (compileClient != null)
+        {
+            SendCompileResponse();
+        }
+    }
+
+    /// <summary>
+    /// Sends the compilation response to the compile client.
+    /// </summary>
+    private static void SendCompileResponse()
+    {
+        try
+        {
+            Debug.Log("CompilationController: Sending compile response.");
```

Comment:
(nit) In the future, since you'll have this depend on RGUnityBots, please use RGDebug instead of Debug - makes it a little easier for us to filter and look at the logs related to our SDK.

Reply:
Switched Debug to RGDebug in this PR.